core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c

```diff
@@ -0,0 +1,556 @@
+			continue;
+
+		memset(&regu_cfg, 0, sizeof(regu_cfg));
+		name = fdt_get_name(fdt, regu_node, NULL);
+
+		regu_cfg.flags |= REGU_BO_FLAG_ENABLE_REGU;
+
+		if (fdt_getprop(fdt, regu_node, "regulator-pull-down", NULL)) {
+			stpmic1_bo_pull_down_cfg(name, &regu_cfg.cfg);
+			regu_cfg.flags |= REGU_BO_FLAG_PULL_DOWN;
+		}
+
+		if (fdt_getprop(fdt, regu_node, "st,mask-reset", NULL)) {
+			stpmic1_bo_mask_reset_cfg(name, &regu_cfg.cfg);
+			regu_cfg.flags |= REGU_BO_FLAG_MASK_RESET;
+		}
+
+		cuint = fdt_getprop(fdt, regu_node,
+				    "regulator-min-microvolt", NULL);
+		if (cuint) {
+			/* DT uses microvolts and driver awaits millivolts */
+			mv = (uint16_t)(fdt32_to_cpu(*cuint) / 1000U);
+
+			if (!stpmic1_bo_voltage_cfg(name, mv, &regu_cfg.cfg))
+				regu_cfg.flags |= REGU_BO_FLAG_SET_VOLTAGE;
+		}
+
+		/* Save config in the Boot On configuration list */
+		regu_bo_count++;
+		regu_bo_config = realloc(regu_bo_config,
+					 regu_bo_count * sizeof(regu_cfg));
+		if (!regu_bo_config)
+			panic();
+
+		memcpy(&regu_bo_config[regu_bo_count - 1], &regu_cfg,
+		       sizeof(regu_cfg));
+	}
+
+	return 0;
+}
+
+void stm32mp_pmic_apply_boot_on_config(void)
+{
+	size_t i = 0;
+
+	for (i = 0; i < regu_bo_count; i++) {
+		struct regu_bo_config *regu_cfg = &regu_bo_config[i];
+
+		if (regu_cfg->flags & REGU_BO_FLAG_SET_VOLTAGE)
+			if (stpmic1_bo_voltage_unpg(&regu_cfg->cfg))
+				panic();
+
+		if (regu_cfg->flags & REGU_BO_FLAG_ENABLE_REGU)
+			if (stpmic1_bo_enable_unpg(&regu_cfg->cfg))
+				panic();
+
+		if (regu_cfg->flags & REGU_BO_FLAG_PULL_DOWN)
+			if (stpmic1_bo_pull_down_unpg(&regu_cfg->cfg))
+				panic();
+
+		if (regu_cfg->flags & REGU_BO_FLAG_MASK_RESET)
+			if (stpmic1_bo_mask_reset_unpg(&regu_cfg->cfg))
+				panic();
+	}
+}
+
+/*
+ * @idx: Private identifier provided by the target PMIC driver
+ * @flags: Operations expected when entering a low power sequence
+ * @voltage: Target voltage to apply during low power sequences
+ */
+struct regu_lp_config {
+	uint8_t flags;
+	struct stpmic1_lp_cfg cfg;
+};
+
+#define REGU_LP_FLAG_LOAD_PWRCTRL	BIT(0)
+#define REGU_LP_FLAG_ON_IN_SUSPEND	BIT(1)
+#define REGU_LP_FLAG_OFF_IN_SUSPEND	BIT(2)
+#define REGU_LP_FLAG_SET_VOLTAGE	BIT(3)
+#define REGU_LP_FLAG_MODE_STANDBY	BIT(4)
+
+struct regu_lp_state {
+	const char *name;
+	size_t cfg_count;
+	struct regu_lp_config *cfg;
+};
+
+enum regu_lp_state_id {
+	REGU_LP_STATE_DISK = 0,
+	REGU_LP_STATE_STANDBY,
+	REGU_LP_STATE_MEM,
+	REGU_LP_STATE_COUNT
+};
+
+static struct regu_lp_state regu_lp_state[REGU_LP_STATE_COUNT] = {
+	[REGU_LP_STATE_DISK] = { .name = "standby-ddr-off", },
+	[REGU_LP_STATE_STANDBY] = { .name = "standby-ddr-sr", },
+	[REGU_LP_STATE_MEM] = { .name = "lp-stop", },
+};
+
+static unsigned int regu_lp_state2idx(const char *name)
+{
+	unsigned int i = 0;
+
+	for (i = 0; i < ARRAY_SIZE(regu_lp_state); i++) {
+		struct regu_lp_state *state = &regu_lp_state[i];
+
+		if (!strncmp(name, state->name, strlen(state->name)))
```

Comment:
When used like this you should rather use `strcmp()`

Reply:
My apologies, it's a comment you already send when reviewing STPMIC1 driver in a previous P-R. Will fix. Thanks.